echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/api/Notification.groovy

```diff
@@ -29,13 +29,15 @@ class Notification {
     String executionType
     String executionId
     String application
+    String user
   }
 
   static enum Type {
     HIPCHAT,
     EMAIL,
     SMS,
-    SLACK
+    SLACK,
+    INCIDENT
```

Comment:
I'm not convinced we want this style of integration to require changes in `echo` + `orca` + `deck` ... which is why I purposely went generic with specific logic only existing at the edge in `echo`. 

I don't see any particular reason why `echo` shouldn't be able to surface failures back to callers ... maybe the `pdApiKey` doesn't exist, or perhaps it does exist but is otherwise invalid.